resen/Resen.py

```diff
@@ -662,13 +662,56 @@ def create_container(self,**input_kwargs):
         if image_id not in local_image_ids:
             print("Pulling image: %s" % image_name)
             print("   This may take some time...")
-            self.docker.images.pull(pull_image)
+            DockerHelper.stream_pull_image(pull_image)
```

Comment:
I don't think this is a good practice and instead the line should be:

    self.stream_pull_image(pull_image)

also, maybe we can have `stream_pull_image` return a status depending on whether it succeeded or not. Then the syntax is:

    status = self.stream_pull_image(pull_image)
    if not status:
        print("ERROR: Failed to pull image")
        return None

Then on line 464, we can have the code check `container_id` like so:

    if container_id is None:
        print("ERROR: Failed to create container')
        return False

or something like that.

Reply:
Sure `stream_pull_image` doesn't need to be a static method. My reasoning was: This def doesn't really use any of the variables or methods of the class, it is independent. So it was a way of saying that. I am sure there are other ways, like having having the definition outside the class. But in favor of getting things moving I'll change that.
I agree on having a return status to check if the image was successfully pulled from the repo. I'll do those changes.